pkg/controller/tas/resource_flavor_test.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tas
+
+import (
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	testingnode "sigs.k8s.io/kueue/pkg/util/testingjobs/node"
+)
+
+func TestNodeHandler_Update(t *testing.T) {
+	now := metav1.Now()
+	later := metav1.NewTime(now.Add(10 * time.Second))
+
+	baseNode := testingnode.MakeNode("test-node").
+		Annotation("test-annotation", "value").
+		Label("topology.kubernetes.io/zone", "zone-a").
+		Label("node-role", "worker").
+		Taints(corev1.Taint{
+			Key:    "test-taint",
+			Value:  "value",
+			Effect: corev1.TaintEffectNoSchedule,
+		}).
+		StatusAllocatable(corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("8"),
+			corev1.ResourceMemory: resource.MustParse("32Gi"),
+		}).
+		StatusConditions(
+			corev1.NodeCondition{
+				Type:               corev1.NodeReady,
+				Status:             corev1.ConditionTrue,
+				LastHeartbeatTime:  now,
+				LastTransitionTime: now,
+			},
+			corev1.NodeCondition{
+				Type:               corev1.NodeMemoryPressure,
+				Status:             corev1.ConditionFalse,
+				LastHeartbeatTime:  now,
+				LastTransitionTime: now,
+			},
+		)
+
+	testCases := map[string]struct {
+		oldNode     *corev1.Node
+		newNode     *corev1.Node
+		wantChanged bool
+	}{
+		"LastHeartbeatTime changed": {
+			oldNode: baseNode.Clone().Obj(),
+			newNode: baseNode.Clone().
+				ConditionHeartbeat(corev1.NodeReady, later).
+				ConditionHeartbeat(corev1.NodeMemoryPressure, later).
+				Obj(),
+			wantChanged: false,
+		},
+		"Annotation changed": {
+			oldNode:     baseNode.Clone().Obj(),
+			newNode:     baseNode.Clone().Annotation("new-annotation", "new-value").Obj(),
+			wantChanged: true,
+		},
+		"Label changed": {
+			oldNode:     baseNode.Clone().Obj(),
+			newNode:     baseNode.Clone().Label("new-label", "new-value").Obj(),
+			wantChanged: true,
+		},
+		"Node Ready status changed": {
+			oldNode: baseNode.Clone().Obj(),
+			newNode: baseNode.Clone().
+				ConditionStatus(corev1.NodeReady, corev1.ConditionFalse, later).
```

Comment:
```suggestion
				StatusConditions(
					corev1.NodeCondition{
						Type:               corev1.NodeReady,
						Status:             corev1.ConditionFalse,
						LastHeartbeatTime:  later,
						LastTransitionTime: later,
					},
				).
```

We want to remove `StatusConditions` call from baseNode, then call that in every test case.
We follow the DUMP principle to make tests as descriptive as possible.

Note: The `ConditionHeartbeat` is still better since it overwrites only one field and is less programmable.